driver/reducecalculation_driver.hpp

```diff
@@ -237,27 +312,51 @@ int ReduceCalculationDriver<Tgpu, Tref>::AllocateBuffersAndCopy()
     uint32_t ctx = 0;
 
     in_dev        = std::unique_ptr<GPUMem>(new GPUMem(ctx, in_sz, sizeof(Tgpu)));
-    out_dev       = std::unique_ptr<GPUMem>(new GPUMem(ctx, out_sz, sizeof(Tgpu)));
+    out_dev       = IsLogicalCalculation() ? std::make_unique<GPUMem>(ctx, out_sz, sizeof(bool))
```

Comment:
why `sizeof(bool)` when you init `out` as `logical_out = std::vector<uint8_t>(out_sz, 0)`

Reply:
it was a mistake. I fixed this